core/src/test/java/org/infinispan/persistence/PersistenceManagerAvailabilityTest.java

```diff
@@ -0,0 +1,107 @@
+package org.infinispan.persistence;
+
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertTrue;
+import static org.testng.AssertJUnit.fail;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.infinispan.Cache;
+import org.infinispan.commons.CacheException;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.global.GlobalConfiguration;
+import org.infinispan.configuration.global.GlobalConfigurationBuilder;
+import org.infinispan.manager.DefaultCacheManager;
+import org.infinispan.notifications.Listener;
+import org.infinispan.notifications.cachelistener.annotation.PersistenceAvailabilityChanged;
+import org.infinispan.notifications.cachelistener.event.PersistenceAvailabilityChangedEvent;
+import org.infinispan.persistence.dummy.DummyInMemoryStore;
+import org.infinispan.persistence.dummy.DummyInMemoryStoreConfigurationBuilder;
+import org.infinispan.persistence.manager.PersistenceManager;
+import org.infinispan.persistence.spi.PersistenceException;
+import org.infinispan.persistence.spi.StoreUnavailableException;
+import org.infinispan.test.AbstractInfinispanTest;
+import org.infinispan.test.TestingUtil;
+import org.infinispan.test.fwk.CleanupAfterMethod;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.testng.annotations.Test;
+
+@CleanupAfterMethod
+@Test(testName = "persistence.PersistenceManagerAvailabilityTest", groups = "functional")
+public class PersistenceManagerAvailabilityTest extends AbstractInfinispanTest {
+
+   private Cache<Object, Object> createManagerAndGetCache(int startFailures) {
+      GlobalConfiguration globalConfiguration = new GlobalConfigurationBuilder().globalJmxStatistics().allowDuplicateDomains(true).build();
+      ConfigurationBuilder cb = TestCacheManagerFactory.getDefaultCacheConfiguration(false);
+      Configuration config = cb.persistence()
```

Comment:
We should lower the availabilityInterval to a smaller number or mock the `ScheduledExecutorService` to run the operation whenever we want in the same thread.